### PR TITLE
Ledger & InboundLedger cleanup

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

File: src/ripple/app/ledger/Ledger.h [hunk 1]
```diff
@@ -98,14 +98,12 @@ class Ledger final
...
-    Ledger (uint256 const& parentHash, uint256 const& transHash,
-            uint256 const& accountHash,
-            std::uint64_t totDrops, NetClock::time_point closeTime,
-            NetClock::time_point parentCloseTime, int closeFlags,
-            NetClock::duration closeResolution,
-            std::uint32_t ledgerSeq, bool & loaded, Config const& config,
-            Family& family,
-            beast::Journal j);
+    Ledger (
+        LedgerInfo info,
```
Comment (HowardHinnant) [inline src/ripple/app/ledger/Ledger.h]: We should pass `LedgerInfo` by `LedgerInfo const&`.  It's too big to pass by value.

File: src/ripple/app/ledger/impl/InboundLedger.cpp [hunk 2]
```diff
@@ -1165,13 +1165,14 @@ void InboundLedger::runData ()
...
+            std::lock_guard<std::recursive_mutex> sl (mReceivedDataLock);
 
             if (mReceivedData.empty ())
             {
                 mReceiveDispatched = false;
                 break;
             }
+
             data.swap(mReceivedData);
         }
```
Comment (HowardHinnant) [inline src/ripple/app/ledger/impl/InboundLedger.cpp]: 👍  At first I didn't see why you didn't get rid of `ScopedLockType`, but I now see that would have a much larger ripple (no pun).

File: src/ripple/app/ledger/impl/InboundLedger.cpp [hunk 3]
```diff
@@ -188,6 +188,31 @@ InboundLedger::neededStateHashes (
...
+    info.drops = sit.get64 ();
+    info.parentHash = sit.get256 ();
+    info.txHash = sit.get256 ();
+    info.accountHash = sit.get256 ();
+    info.parentCloseTime = NetClock::time_point{NetClock::duration{sit.get32()}};
+    info.closeTime = NetClock::time_point{NetClock::duration{sit.get32()}};
+    info.closeTimeResolution = NetClock::duration{sit.get8()};
+    info.closeFlags = sit.get8 ();
+
+    return std::move(info);
```
Comment (HowardHinnant) [inline src/ripple/app/ledger/impl/InboundLedger.cpp]: I see the move got removed later! 👍

File: src/ripple/app/ledger/Ledger.cpp [hunk 4]
```diff
@@ -189,9 +207,7 @@ Ledger::Ledger (create_genesis_t, Config const& config, Family& family)
     sle->setFieldAmount (sfBalance, info_.drops);
     rawInsert(sle);
     stateMap_->flushDirty (hotACCOUNT_NODE, info_.seq);
-    updateHash();
     setImmutable(config);
-    setup(config);
```
Comment (HowardHinnant) [inline src/ripple/app/ledger/Ledger.cpp]: For my own education:  Why are `updateHash()` and `setup()` not needed here?

Comment (HowardHinnant) [inline src/ripple/app/ledger/Ledger.cpp]: Never mind, I see it in `setImmutable` now! :-)

**Comment by HowardHinnant** [comment]
Nice!  I have only one minor comment with an action item on it (the one about passing `LedgerInfo` by value).

File: src/ripple/app/ledger/Ledger.h (hunk 1 above)
Comment (nbougalis) [inline src/ripple/app/ledger/Ledger.h]: Done

**Comment by HowardHinnant** [comment]
👍

File: src/ripple/app/ledger/impl/InboundLedger.cpp (hunk 2 above)
Comment (vinniefalco) [inline src/ripple/app/ledger/impl/InboundLedger.cpp]: @HowardHinnant  http://www.thebestpageintheuniverse.net/c.cgi?u=puns

**Comment by nbougalis** [comment]
Rebased, addressed concerns and squashed.

**Comment by nbougalis** [comment]
Rebased to b11

**Comment by JoelKatz** [comment]
:+1:

**Pull request closed without merging**